apps/worker/src/monitor/tx.status.monitor.service.ts

```diff
@@ -34,42 +35,44 @@ export class TxStatusMonitoringService extends BaseConsumer {
       const previousKnownBlockNumber = (await this.blockchainService.getBlock(job.data.lastFinalizedBlockHash)).block.header.number.toBigInt();
       const currentFinalizedBlockNumber = await this.blockchainService.getLatestFinalizedBlockNumber();
       const blockList: bigint[] = [];
-      const blockDelay = 1 * SECONDS_PER_BLOCK * MILLISECONDS_PER_SECOND;
 
       for (let i = previousKnownBlockNumber; i <= currentFinalizedBlockNumber && i < previousKnownBlockNumber + numberBlocksToParse; i += 1n) {
         blockList.push(i);
       }
-      const txResult = await this.blockchainService.crawlBlockListForTx(job.data.txHash, blockList, [{ pallet: 'messages', event: 'MessagesStored' }]);
+      const txResult = await this.blockchainService.crawlBlockListForTx(job.data.txHash, blockList, [{ pallet: 'system', event: 'ExtrinsicSuccess' }]);
 
-      this.setEpochCapacity(txCapacityEpoch, BigInt(txResult.capacityWithDrawn ?? 0n));
-
-      if (txResult.success && txResult.blockHash && !txResult.error) {
-        this.logger.verbose(`Successfully found ${job.data.txHash} found in block ${txResult.blockHash}`);
-        return txResult;
-      }
-
-      // if tx has not yet included in a block, throw error to retry till max attempts
-      if (!txResult.blockHash && !txResult.error) {
-        throw new Error(`Tx not found in block list, retrying (attempts=${job.attemptsMade})`);
-      }
+      if (!txResult.found) {
+        if (job.attemptsMade < (job.opts.attempts ?? 3)) {
+          // if tx has not yet included in a block, throw error to retry till max attempts
+          throw new Error(`Tx not found in block list, retrying (attempts=${job.attemptsMade})`);
+        } else {
+          this.logger.warn(`Could not fetch the transaction adding to publish again! ${job.id}`);
+          // could not find the transaction, this might happen if transaction never gets into a block
+          await this.retryPublishJob(job.data.referencePublishJob);
+        }
+      } else {
+        // found the tx
+        await this.setEpochCapacity(txCapacityEpoch, BigInt(txResult.capacityWithDrawn ?? 0n));
```

Comment:
we probably need a better way of calculating capacity since there might be other transactions that we don't care but affect capacity

Reply:
I think we can leverage the fact that if we find capacity withdrawn event after we have discovered an event that we are looking for, the very next event from record we would expect to be `capacityWithDrawn` 

However these are good metrics for sure, 

Currenrly for actual use of tracking total consumer and available capacity helps to take action before publishing ( not sure we are doing it )

Reply:
But definitely we can constraint the value to only our ext if we want it before release

Reply:
Watching capacity can actually be an isolated module which we might be part of the chain watcher project.